datastore/query.go

```diff
@@ -171,25 +177,47 @@ func (q *Query) Transaction(t *Transaction) *Query {
 }
 
 // Filter returns a derivative query with a field-based filter.
+//
+// DEPRECATED. Use the FilterWithArgs() method instead, which supports the same
+// set of operations (and more).
+//
 // The filterStr argument must be a field name followed by optional space,
-// followed by an operator, one of ">", "<", ">=", "<=", or "=".
+// followed by an operator, one of ">", "<", ">=", "<=", "=", and "!=".
 // Fields are compared against the provided value using the operator.
 // Multiple filters are AND'ed together.
 // Field names which contain spaces, quote marks, or operator characters
 // should be passed as quoted Go string literals as returned by strconv.Quote
 // or the fmt package's %q verb.
 func (q *Query) Filter(filterStr string, value interface{}) *Query {
+	// TODO( #5977 ): Add better string parsing (or something)
 	q = q.clone()
```

Comment:
Looks like this clone can be removed if you are cloning in FilterWithArgs as well.

Reply:
Good point. Done.